x-pack/plugins/search_playground/public/components/app.tsx

```diff
@@ -5,42 +5,85 @@
  * 2.0.
  */
 
-import React, { useEffect, useState } from 'react';
+import React, { useState } from 'react';
 import { KibanaPageTemplate } from '@kbn/shared-ux-page-kibana-template';
 
 import { useWatch } from 'react-hook-form';
 import { QueryMode } from './query_mode/query_mode';
-import { SetupPage } from './setup_page/setup_page';
+import { ChatSetupPage } from './setup_page/chat_setup_page';
 import { Header } from './header';
 import { useLoadConnectors } from '../hooks/use_load_connectors';
-import { ChatForm, ChatFormFields } from '../types';
+import { ChatForm, ChatFormFields, PlaygroundPageMode } from '../types';
 import { Chat } from './chat';
+import { SearchMode } from './search_mode/search_mode';
+import { SearchPlaygroundSetupPage } from './setup_page/search_playground_setup_page';
+import { usePageMode } from '../hooks/use_page_mode';
 
 export interface AppProps {
   showDocs?: boolean;
+  pageMode?: PlaygroundPageMode;
 }
 
 export enum ViewMode {
   chat = 'chat',
   query = 'query',
 }
 
-export const App: React.FC<AppProps> = ({ showDocs = false }) => {
-  const [showSetupPage, setShowSetupPage] = useState(true);
+export const App: React.FC<AppProps> = ({
+  showDocs = false,
+  pageMode = PlaygroundPageMode.chat,
+}) => {
   const [selectedMode, setSelectedMode] = useState<ViewMode>(ViewMode.chat);
   const { data: connectors } = useLoadConnectors();
-  const hasSelectedIndices = useWatch<ChatForm, ChatFormFields.indices>({
-    name: ChatFormFields.indices,
-  }).length;
-  const handleModeChange = (id: string) => setSelectedMode(id as ViewMode);
+  const hasSelectedIndices = Boolean(
+    useWatch<ChatForm, ChatFormFields.indices>({
```

Comment:
did you try getValues instead? https://react-hook-form.com/docs/useform/getvalues

Reply:
Well, I tried after your comment I am not sure why but cannot return results from there 🤔